### PR TITLE
Fix for the delete button in the query field

### PR DESCRIPTION
**Pull request by prahlad37**
delete button is not working in the query field.
This patch is tested on command line ubuntu.

Signed-off-by: Prahlad V <prahlad.eee@gmail.com>

### CODE REVIEW

**Comment by calleerlandsson** [comment]
Hey @prahlad37, thanks for contributing! Am I understanding correctly that what you would like is for the delete key to work the same as the Ctrl-D key combination does? I.e. delete the character under the cursor not to the left of it.

I tried your branch locally on OS X 10.11.2 and could reliably reproduce segmentation faults when pressing the delete key when the query field is empty or when currently at the last column of the query.

What do you think about instead introducing another fall through case above the `CTRL('D')` one with the right key for delete?

**Comment by prahlad37** [comment]
My bad.. I didn't see the CTRL+D binding. I think it should be fine to add it as a fall through case of CTRL(D)

**Comment by calleerlandsson** [comment]
Hey @prahlad37, that looks much better! I'm kind of confused by the naming of these constants. You're moving `BACKSPACE` but we're talking about the delete key. Should these names be swapped?

@mptre Do you have more background knowledge on these constants?

**Comment by mptre** [comment]
Pressing the delete key in my shell outputs the key_dc (abbreviated kD)
capability which is equal to the sequence present in this patch. The
behavior of this key is equivalent to Ctrl-D so I think this change
makes sense.

On Fri, Apr 29, 2016 at 07:44:57AM -0700, Calle Erlandsson wrote:
> Hey @prahlad37, that looks much better! I'm kind of confused by the naming of these constants. You're moving `BACKSPACE` but we're talking about the delete key. Should these names be swapped?

Agree, they should be swapped.

A few other notes on the patch:

- Don't introduce whitespace changes, referring to the deleted blank line

- Please add a test-case for the new key binding, the test should look
  similiar to tests/03.in

- Swap the DEL and BACKSPACE constants as discussed

- Squash your commits once you're done

Other than that, great work.

File: src/pick.c [hunk 1]
```diff
@@ -356,6 +357,7 @@ selected_choice(void)
 			}
 
 			break;
+		case DEL:
```
Comment (mptre) [inline src/pick.c]: Only one case label is needed here. I suggest you remove the `CTRL_D` value completely and let the `CTRL('D')` binding return `DEL` instead.

File: tests/18.in [hunk 2]
```diff
@@ -0,0 +1,6 @@
+description: move backward and delete character under cursor
```
Comment (mptre) [inline tests/18.in]: Would be great if the delete key was mentioned explicitly here.

**Comment by mptre** [comment]
Latest changes looks good. One last thing: don't forget to add the new
key to the man page.

**Comment by calleerlandsson** [comment]
Hey @prahlad37, thanks for working on this! I think the latest changes look great.

The tests fail locally for me with this error message:

```
./tests/test.sh: line 41: [: too many arguments
./tests/10.in: wrong exit code
test: time limit exceeded
FAIL tests/test.sh (exit status: 1)
```

Do you see something similar when you run `make check; cat test-suite.log`?

**Comment by calleerlandsson** [comment]
With #113 merged, I now get this when running `make check; cat test-suite.log`.

```
./tests/10.in: wrong exit code
test: time limit exceeded
FAIL tests/test.sh (exit status: 1)
```

**Comment by prahlad37** [comment]
I am also having the same problem.
After debugging arrived to a point that the default canonical mode of master tty is cause processing of our input before feeding to pick. #119 looks like the possible fix.

**Comment by calleerlandsson** [comment]
After merging #119 and rebasing on top of master `make check` passes! Thanks for working on this @prahlad37, I'll merge this PR.

**Comment by calleerlandsson** [comment]
Merged as ac92b24d9c57f6327061dd18b1f2daa97559226b.

**Pull request closed without merging**